### ISSUE TITLE
Poll privacy

### ISSUE THREAD

**Issue by petrenkonikita112263** [xwikisas/application-xpoll#105]
Currently, anyone who has view rights for the poll can see users' votes and result. It'll be better to enhance poll privacy - where only the author (creator) of a poll can see votes and results. Or allow the user to see his/her own votes and doesn't see other users' votes.

**Comment by lucaa** [comment]
This should be an option, it should not be the default access schema for the polls.

**Comment by snazare** [comment]
Hi @oanalavinia indeed, I agree with your suggestion - we should enforce the edit right only for the poll creator. Other users should not be able to see the votes in edit mode as this would defeat the purpose of the private option.

**Comment by petrenkonikita112263** [comment]
>Use case I am referring: I create a poll which is private (only I can see the results). A user with edit rights will access the poll and would only see his votes in the UI. But he can still open the poll in object mode and see votes (cannot see the result computation, but can do this manually)

exactly! but it's only in case if those users are experienced users which isn't common

>This would make some things easier maybe: we will display the poll votes for those users with edit rights, since maybe the creator would want to pass this view role to other users as well (I created a private poll, but I want some other users to be able to see the results)

plus for this approach

**Comment by snazare** [comment]
On this ticket we will work on adding a parameter in the poll page, to make all votes public (anyone that can see the poll can see everyone's votes and the results), or make votes private (current user can only see, add and change their vote; only creator of the poll can see all votes and results).
